src/openrct2/scripting/ScTile.hpp

```diff
@@ -0,0 +1,1112 @@
+
+#pragma once
+
+#ifdef __ENABLE_SCRIPTING__
+
+#    include "../Context.h"
+#    include "../common.h"
+#    include "../world/Footpath.h"
+#    include "../world/Scenery.h"
+#    include "../world/Sprite.h"
+#    include "../world/Surface.h"
+#    include "Duktape.hpp"
+#    include "ScriptEngine.h"
+
+#    include <cstdio>
+#    include <cstring>
+
+namespace OpenRCT2::Scripting
+{
+    class ScSurfaceElement;
+
+    class ScTileElement
+    {
+    protected:
+        CoordsXY _coords;
+        TileElement* _element;
+
+    public:
+        ScTileElement(CoordsXY coords, TileElement* element)
+            : _coords(coords)
+            , _element(element)
+        {
+        }
+
+    private:
+        std::string type_get() const
+        {
+            switch (_element->GetType())
+            {
+                case TILE_ELEMENT_TYPE_SURFACE:
+                    return "surface";
+                case TILE_ELEMENT_TYPE_PATH:
+                    return "footpath";
+                case TILE_ELEMENT_TYPE_TRACK:
+                    return "track";
+                case TILE_ELEMENT_TYPE_SMALL_SCENERY:
+                    return "small-scenery";
+                case TILE_ELEMENT_TYPE_ENTRANCE:
+                    return "entrance";
+                case TILE_ELEMENT_TYPE_WALL:
+                    return "wall";
+                case TILE_ELEMENT_TYPE_LARGE_SCENERY:
+                    return "large-scenery";
+                case TILE_ELEMENT_TYPE_BANNER:
+                    return "banner";
+                case TILE_ELEMENT_TYPE_CORRUPT:
+                    return "openrct2-corrupt";
+                default:
+                    return "unknown";
+            }
+        }
+
+        void type_set(std::string value)
+        {
+            auto type = _element->type;
+            if (value == "surface")
+                type = TILE_ELEMENT_TYPE_SURFACE;
+            else if (value == "footpath")
+                type = TILE_ELEMENT_TYPE_PATH;
+            else if (value == "track")
+                type = TILE_ELEMENT_TYPE_TRACK;
+            else if (value == "small-scenery")
+                type = TILE_ELEMENT_TYPE_SMALL_SCENERY;
+            else if (value == "entrance")
+                type = TILE_ELEMENT_TYPE_ENTRANCE;
+            else if (value == "wall")
+                type = TILE_ELEMENT_TYPE_WALL;
+            else if (value == "large-scenery")
+                type = TILE_ELEMENT_TYPE_LARGE_SCENERY;
+            else if (value == "banner")
+                type = TILE_ELEMENT_TYPE_BANNER;
+            else if (value == "openrct2-corrupt")
+                type = TILE_ELEMENT_TYPE_CORRUPT;
+            else
+                return;
+            _element->type = type;
+            map_invalidate_tile_full(_coords);
+        }
+
+        uint8_t baseZ_get() const
+        {
+            return _element->base_height;
+        }
+        void baseZ_set(uint8_t newBaseHeight)
```

Comment:
These names are a bit confusing, given that we have `GetBaseZ()` and it's an order of magnitude greater than `base_height`

Note that if you change it, you have to change the return type.